### DISCUSSION TITLE
Why v13-dev requires some files in runtime

### DISCUSSION THREAD

**Discussion by Drylozu** [discordjs/discord.js#6070] [Q&A] [answered]
I think the title says everything but there's [an example](https://github.com/discordjs/discord.js/blob/master/src/structures/interfaces/TextBasedChannel.js#L152-L154).
There're some `require`s inside the `send` function.

Isn't it bad to require files within functions? Shouldn't it be at the root of the file?

**Comment by iShibi** [comment]
The code you're referring to was last touched three years. It's not that it was added for v13, it's more of something that hasn't been changed.

**Comment by monbrey** [comment, marked as answer]
This is intentional - scoping the require avoids creating circular imports.

**Reply by kyranet** [reply]
It could be moved to the bottom, which also fixes the circular:

https://github.com/discordjs/discord.js/blob/755c180659c125532fe6f8e33e6c3b56e275311b/src/structures/interfaces/TextBasedChannel.js#L428-L431

Circulars only happen because A requires B, B requires A, and A's exports weren't available, which is why that require is below the export.
